apps/sms/views/shared/js/localization_helper.js

```diff
@@ -11,12 +11,20 @@
       'iframe.attachment-container'
     );
 
-    Array.forEach(attachmentContainers, function(attachmentContainer) {
-      var doc = attachmentContainer.contentDocument;
-      doc.documentElement.lang = navigator.mozL10n.language.code;
-      doc.documentElement.dir = navigator.mozL10n.language.direction;
-      navigator.mozL10n.translateFragment(doc.body);
-    });
+    return document.l10n.resolvedLanguages().then(
+      ([lang]) => Promise.all(
+        Array.map(attachmentContainers,
+          container => localizeAttachmentIFrame(container, lang))));
```

Comment:
Please don't use Array generics as it's not standard and might get removed (source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#Array_generic_methods).

Better use Array.from :)
(I know we had Array.forEach before...)

Another nit is that our indentation convention is quite different. So the final code could look like this:
```javascript
var attachmentContainers = Array.from(document.querySelectorAll(
  'iframe.attachment-container'
));

return document.l10n.resolvedLanguages().then(
  ([lang]) => Promise.all(
    attachmentContainers.map(
      container => localizeAttachmentIFrame(container, lang)
    )
  )
);
```

Basically, you can have 2 closing parenthesises on one line, if they were opened on the same line.